src/topp/PercolatorAdapter.cpp

```diff
@@ -0,0 +1,1026 @@
+    registerStringOption_("default-direction", "<featurename>", "", "The most informative feature given as the feature name, can be negated to indicate that a lower value is better.", !is_required, is_advanced_option);
+    registerIntOption_("verbose", "<level>", 2, "Set verbosity of output: 0=no processing info, 5=all.", !is_required, is_advanced_option);
+    registerFlag_("unitnorm", "Use unit normalization [0-1] instead of standard deviation normalization", is_advanced_option);
+    registerFlag_("test-each-iteration", "Measure performance on test set each iteration", is_advanced_option);
+    registerFlag_("override", "Override error check and do not fall back on default score vector in case of suspect score vector", is_advanced_option);
+    registerIntOption_("seed", "<value>", 1, "Setting seed of the random number generator.", !is_required, is_advanced_option);
+    registerIntOption_("doc", "<value>", 0, "Include description of correct features", !is_required, is_advanced_option);
+    registerFlag_("klammer", "Retention time features calculated as in Klammer et al. Only available if -doc is set", is_advanced_option);
+    registerInputFile_("fasta", "<file>", "", "Provide the fasta file as the argument to this flag, which will be used for protein grouping based on an in-silico digest (only valid if option -protein-level-fdrs is active).", !is_required, is_advanced_option);
+    setValidFormats_("fasta", ListUtils::create<String>("FASTA"));
+    registerStringOption_("decoy-pattern", "<value>", "random", "Define the text pattern to identify the decoy proteins and/or PSMs, set this up if the label that identifies the decoys in the database is not the default (Only valid if option -protein-level-fdrs is active).", !is_required, is_advanced_option);
+    registerFlag_("post-processing-tdc", "Use target-decoy competition to assign q-values and PEPs.", is_advanced_option);
+  }
+  
+  // TODO replace with TopPerc::getScanMergeKey
+  String getScanIdentifier_(vector<PeptideIdentification>::iterator it, vector<PeptideIdentification>::iterator start)
+  {
+    // MSGF+ uses this field, is empty if not specified
+    String scan_identifier = it->getMetaValue("spectrum_reference");
+    if (scan_identifier.empty())
+    {
+      // XTandem uses this (integer) field
+      // these ids are 1-based in contrast to the index which is 0-based. This might be problematic to use for merging
+      if (it->metaValueExists("spectrum_id") && !it->getMetaValue("spectrum_id").toString().empty())
+      {
+        scan_identifier = "scan=" + it->getMetaValue("spectrum_id").toString();
+      }
+      else
+      {
+        scan_identifier = "index=" + String(it - start + 1);
+        LOG_WARN << "no known spectrum identifiers, using index [1,n] - use at own risk." << endl;
+      }
+    }
+    return scan_identifier.removeWhitespaces();
+  }
+  
+  // TODO replace with TopPerc::getScanMergeKey
```

Comment:
TopPerc::getScanMergeKey has to be split in two to yield that functionality though.